pkg/entities/set.go

```diff
@@ -99,13 +109,14 @@ func (s *set) PrepareSet(setType ContentType, templateID uint16) error {
 
 func (s *set) ResetSet() {
 	if !s.isDecoding {
-		s.headerBuffer = nil
-		s.headerBuffer = make([]byte, SetHeaderLen)
+		clear(s.headerBuffer)
 		s.length = SetHeaderLen
 	}
 	s.setType = Undefined
-	s.records = nil
-	s.records = make([]Record, 0)
+	// Clear before shrinking the slice so that existing elements are eligible for garbage collection.
```

Comment:
Do we need to reset the s.length here? I'm not quite sure what's the case when it isDecoding.

Reply:
I can reset it to 0, even though it's a change to the existing behavior. I agree that it would make more sense.
I am not a big fan of the `isDecoding` flag everywhere. We should have `Marhsal` / `Unmarshal` methods as appropriate, but that will have to be for a later time...